centovacast.php

```diff
@@ -1826,13 +1831,37 @@ private function getFieldsFromInput(array $vars, $package)
         ];
 
         $fields = array_merge((array) $package->meta, $fields);
-
-        if ($fields['diskquota'] == 0) {
-            $fields['diskquota'] = 'unlimited';
-        }
-        if ($fields['transferlimit'] == 0) {
-            $fields['transferlimit'] = 'unlimited';
-        }
+		
+		//configoptions (Configurable Options)
+		$fieldsConfigOptions = array_merge((array) $package->meta, $fields['configoptions']);
```

Comment:
$fields['configoptions'] is not defined, and if you mean to use $vars['configoptions'] then you should check if it exists as we do above for other fields.

That being said, you don't even need to do this merge.  Your intent seems to be that you want to use the config option if it exist, otherwise use the package meta.  In that case you can use $vars['configoptions']['diskquota'] in the places you are using fieldsConfigOptions['diskquota'].  It will have the same effect. The same applies to transfer limit